app/_layout.tsx

```diff
@@ -1,28 +1,55 @@
-import { Suspense } from "react";
+import { Suspense, useCallback, useEffect, useState } from "react";
 import { useColorScheme } from "react-native";
 import {
   DarkTheme,
   DefaultTheme,
   ThemeProvider
 } from "@react-navigation/native";
-import { useFonts } from "expo-font";
 import { SplashScreen, Stack } from "expo-router";
-import { StatusBar } from "expo-status-bar";
 import { TamaguiProvider, Text, Theme } from "tamagui";
 
 import { MySafeAreaView } from "../components/MySafeAreaView";
+import useFonts from "../hooks/useFonts";
 import config from "../tamagui.config";
 
+SplashScreen.preventAutoHideAsync();
+
 export default function Layout() {
   const colorScheme = useColorScheme();
+  const [appIsReady, setAppIsReady] = useState(false);
+
+  useEffect(() => {
+    async function prepare() {
+      try {
+        // Pre-load fonts, make any API calls you need to do here
+        await useFonts();
+        // Artificially delay for two seconds to simulate a slow loading
+        // experience.
+        // await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
do we want to do this ?

Reply:
Could be useful for when we're testing. For now I left it as a comment